frontend/cypress/integration/common/traffic_policies_multicluster.ts

```diff
@@ -0,0 +1,57 @@
+import { Then, When } from '@badeball/cypress-cucumber-preprocessor';
+import { ensureKialiFinishedLoading } from './transition';
+
+const authorizationPolicies: string[] = ['deny-all-bookinfo', 'details-v1', 'productpage-v1','ratings-v1','reviews-v1','reviews-v2','reviews-v3'];
+const sidecars: string[] = ['kiali-traffic-generator', 'details-v1', 'productpage-v1','ratings-v1','reviews-v1','reviews-v2','reviews-v3'];
+
+When('user deletes a Traffic Policy and the resource is no longer available in any cluster',() => {
+  authorizationPolicies.forEach((policy) => {
+    cy.request({failOnStatusCode:false,
+                url:`/api/namespaces/bookinfo/istio/authorizationpolicies/${policy}?clusterName=east`,
+                method:'DELETE'});
+    cy.request({failOnStatusCode:false,
+                url:`/api/namespaces/bookinfo/istio/authorizationpolicies/${policy}?clusterName=west`,
+                method:'DELETE'});
+  });
+  sidecars.forEach((sidecar) => {
+  cy.request({failOnStatusCode:false,
+                url:`/api/namespaces/bookinfo/istio/sidecars/${sidecar}?clusterName=east`,
+                method:'DELETE'});
+    cy.request({failOnStatusCode:false,
+                url:`/api/namespaces/bookinfo/istio/sidecars/${sidecar}?clusterName=west`,
+                method:'DELETE'});
+  });
+});
+
+When('user decides to {string} a Traffic Policy in the {string} {string}',(action:string,cluster:string,ns:string) =>{
+  cy.get(`[data-test="CardItem_${ns}_${cluster}"]`).find('[aria-label="Actions"]').click();
+  cy.get('button').contains(`${action[0].toUpperCase() + action.slice(1)} Traffic Policies`).should('be.visible');
+  cy.get('button').contains(`${action[0].toUpperCase() + action.slice(1)} Traffic Policies`).click();
+  ensureKialiFinishedLoading();
+});
+
+When('user confirms to {string} the Traffic Policy',(action:string) => {
+  if (action === 'create' || action === 'update'){
+    cy.get(`button[data-test="${action}"]`).click();
+  }
+  cy.get('button[data-test="confirm-create"]').click();
```

Comment:
is it hardcoded 'create' for update as well?

Reply:
Hello. Yes it is, but it only applies for this specific modal on the Overview page I think.